sources/documents/2_RTB/Verbali/Esterni/VerbaleEsterno_2024-05-06/capitolo-riunione.tex

```diff
@@ -0,0 +1,59 @@
+
+\subsubsection{Esposizione studio preliminare delle tecnologie lato \glossario{back-end}}
+
+\textbf{Descrizione:} Il \glossario{Programmatore} espone il lavoro di studio sulla libreria \glossario{txtai}, con particolare enfasi riguardo la componente di indicizzazione e ricerca. 
+Pone poi l'attenzione sulla necessità o meno di fasi di pre-processing delle queries, nel tentativo di semplificare la ricerca semantica per la LLM. 
+Questo riguarda l'elaborazione della query utente in modo tale da tokenizzarne solo le componenti semantiche ritenute fondamentali, e allo stesso modo l'elaborazione del dizionario dati in formato JSON per renderlo simile al linguaggio naturale, in modo che i modelli, allenati su quest'ultimo, riescano ad estrarne con più facilità il significato.
+
+\textbf{Risposta: }La \glossario{Proponente} ritiene le fasi di pre-processing non necessarie, in quanto il dizionario dati risulta sufficiente per creare degli embeddings efficaci ed associabili alle query utente senza ulteriori alterazioni. 
+Dimostra poi l'utilizzo di txtai per la ricerca semantica tramite BERTino di efederici, un modello operante su lingua italiana, ed invita all'utilizzo delle funzioni di load e save per ridurre l'overhead e gestire con più facilità il testing di più indici.
+
+\textbf{Descrizione:} Il \glossario{Programmatore} espone il lavoro di studio sulla libreria txtai, con particolare enfasi riguardo la componente di indicizzazione e ricerca. 
+Pone poi l'attenzione sulla necessità o meno di fasi di pre-processing delle queries, nel tentativo di semplificare la ricerca semantica per la LLM. 
+Questo riguarda l'elaborazione della query utente in modo tale da tokenizzarne solo le componenti semantiche ritenute fondamentali, e allo stesso modo l'elaborazione del dizionario dati in formato JSON per renderlo simile al linguaggio naturale, in modo che i modelli, allenati su quest'ultimo, riescano ad estrarne con più facilità il significato.
+
+\textbf{Risposta: }La \glossario{Proponente} ritiene le fasi di pre-processing non necessarie, in quanto il dizionario dati risulta sufficiente per creare degli embeddings efficaci ed associabili alle query utente senza ulteriori alterazioni. 
+Dimostra poi l'utilizzo di txtai per la ricerca semantica tramite BERTino di efederici, un modello operante su lingua italiana, ed invita all'utilizzo delle funzioni di load e save per ridurre l'overhead e gestire con più facilità il testing di più indici.
+
+\textbf{Descrizione:} Il \glossario{Programmatore} espone il problema dello scarso score dei risultati restituiti dalla funzione di search di txtai, e suggerisce la mitigazione del problema tramite un indice intermedio che si frapponga alla query utente e al risultato finale da fornire a una LLM come ChatGPT per ottenere l'effettiva query SQL.
+
+\textbf{Risposta: }La \glossario{Proponente} accoglie l'idea e ne invita la sperimentazione. 
```

Comment:
\textbf{Risposta:} La \glossario{Proponente} accoglie la proposta e ne invita la sperimentazione. 
Riguardo allo score basso degli output forniti dalla search, la Proponente suggerisce l'esplorazione di più modelli. 
Spiega inoltre che txtai utilizza, in maniera del tutto trasparente, la libreria \glossario{FAISS}, che si occupa della \glossario{vettorizzazione} e indicizzazione oltre che della ricerca semantica. In aggiunta, la Proponente suggerisce di confrontare FAISS con altre librerie.